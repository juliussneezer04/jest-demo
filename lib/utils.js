/**
 * Divides two numbers and returns the result.
 */
export function divideNumbers(numerator, denominator) {
  if (denominator === 0) {
    throw new Error("Cannot divide by zero");
  }
  return numerator / denominator;
}
