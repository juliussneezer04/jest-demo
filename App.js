import { StatusBar } from "expo-status-bar";
import React, { useState } from "react";
import { StyleSheet, Text, TextInput, View } from "react-native";
import { divideNumbers } from "./lib/utils";

export default function App() {
  const [numerator, setNumerator] = useState(1);
  const [denominator, setDenominator] = useState(1);
  return (
    <View style={styles.container}>
      <View style={styles.line}>
        <Text style={styles.label}>Numerator: </Text>
        <Text style={styles.label}>{numerator}</Text>
      </View>
      <View style={styles.line}>
        <TextInput
          onChangeText={(t) =>
            t === "" ? setNumerator(1) : setNumerator(parseInt(t))
          }
          placeholder="Enter New Numerator"
          style={[styles.label, styles.input]}
        />
      </View>
      <View style={styles.line}>
        <Text style={styles.label}>Denominator: </Text>
        <Text style={styles.label}>{denominator}</Text>
      </View>
      <View style={styles.line}>
        <TextInput
          onChangeText={(t) =>
            t === "" ? setDenominator(1) : setDenominator(parseInt(t))
          }
          placeholder="Enter New Denominator"
          style={[styles.label, styles.input]}
        />
      </View>
      <View style={styles.line}>
        <Text style={styles.label}>Division Result: </Text>
        <Text style={styles.label}>
          {divideNumbers(numerator, denominator)}
        </Text>
      </View>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    rowGap: 10,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  line: {
    display: "flex",
    flexDirection: "row",
    gap: 8,
    alignItems: "center",
  },
  label: {
    fontSize: 24,
    fontWeight: "bold",
  },
  input: {
    backgroundColor: "lightgray",
    padding: 8,
  },
});
