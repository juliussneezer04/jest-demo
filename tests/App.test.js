import React from "react";
import renderer from "react-test-renderer";

import App from "../App";
import { divideNumbers } from "../lib/utils";

describe("<App />", () => {
  it("has 5 child", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(5);
  });
});


describe("Division Function", () => {
  // NOTE: Use %d for numbers, %s for strings, %j for JSON
  it("can divide non-zero numbers", () => {
    const num = 10;
    const denom = 2;
    const expected = 5;
    const res = divideNumbers(num, denom);
    expect(res).toBe(expected);
  });

  
  it("throws an error when dividing by zero", () => {
    expect(() => divideNumbers(1, 0)).toThrow();
  });
});
