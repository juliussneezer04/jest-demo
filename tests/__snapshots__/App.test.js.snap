// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> can render 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "rowGap": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Numerator: 
    </Text>
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter New Numerator"
      style={
        [
          {
            "fontSize": 24,
            "fontWeight": "bold",
          },
          {
            "backgroundColor": "lightgray",
            "padding": 8,
          },
        ]
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Denominator: 
    </Text>
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter New Denominator"
      style={
        [
          {
            "fontSize": 24,
            "fontWeight": "bold",
          },
          {
            "backgroundColor": "lightgray",
            "padding": 8,
          },
        ]
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Division Result: 
    </Text>
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      42
    </Text>
  </View>
</View>
`;
